stackslib/src/net/p2p.rs

```diff
@@ -3874,6 +3865,145 @@ impl PeerNetwork {
         download_backpressure: bool,
         ibd: bool,
         network_result: &mut NetworkResult,
+    ) -> bool {
+        let cur_epoch = self.get_current_epoch();
+        let prune = if cur_epoch.epoch_id >= StacksEpochId::Epoch30 {
+            debug!("{:?}: run Nakamoto work loop", self.get_local_peer());
+
+            // in Nakamoto epoch, so do Nakamoto things
+            let prune = self.do_network_work_nakamoto(sortdb, ibd);
+
+            // in Nakamoto epoch, but we might still be doing epoch 2.x things since Nakamoto does
+            // not begin on a reawrd cycle boundary.
+            if self.burnchain_tip.block_height <= cur_epoch.start_height
```

Comment:
Should this be explicitly the `Epoch30` boundary? Presumably, a hypothetical `Epoch31` wouldn't want to do Epoch 2.x work during `Epoch30`

Reply:
Yes; I'll constrain this check